pymongo/network.py

```diff
@@ -267,7 +269,9 @@ def receive_message(
         data = _receive_data_on_socket(sock_info, length - 16, deadline)
 
     try:
-        unpack_reply = _UNPACK_REPLY[op_code]
+        unpack_reply: Callable[[bytes], _OpReply] = cast(
```

Comment:
How about let's add these type hints to _UNPACK_REPLY? Then we can remove the cast here.